### PR TITLE
Non-obstructive navbar toggle icon (updated)

### PR DESCRIPTION
**Pull request by kapito**
Changes the design of the navbar toggle icon so that it no longer obstructs page headers.

[edit: see images below]

### CODE REVIEW

**Comment by jjanssen** [comment]
Hi @kapito I'm in the midst of reviewing but it seems like the last commit is a part of the #3705

**Comment by kapito** [comment]
@jjanssen Thanks for letting me know -- not sure how I did that! I'll remove it now

**Comment by kapito** [comment]
Thanks for reviewing it @jjanssen -- I agree that it will probably look better if we simply add padding to the left of the headers (as well as being slightly easier to click/touch on mobile, as with my changes it was rather narrow). I'll try to implement/test these changes later.

**Comment by kapito** [comment]
I added some code to _header.scss to increase the header-title padding by 50px when Wagtail is in its mobile state, and also removed my original code from _main-nav.scss. Seems to have done the trick and I'll admit it looks much cleaner than my initial proposed changes. Here's what it looks like now:

image

(I also added `top: 4px;` to _main-nav.scss so that the toggle button aligns correctly with the header title next to it; if there is another file this should be done in please say.)

**Comment by kapito** [comment]
Also fixed success/error/warning messages, which were being obstructed in the same way:

image

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss [hunk 1]
```diff
@@ -160,3 +155,10 @@ header {
         }
     }
 }
+
+// make way for the navbar toggle icon when in mobile state
+@media screen and (max-width: $breakpoint-mobile) {
+    .header-title {
+        padding-left: 50px;
```
Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss]: Can we make a variable for this? Maybe something ``$mobile-indent``?

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_main-nav.scss [hunk 2]
```diff
@@ -473,3 +474,10 @@ body.explorer-open {
         }
     }
 }
+
+// make way for the navbar toggle icon when in mobile state
+@media screen and (max-width: $breakpoint-mobile) {
+    .avatar {
+        left: 50px;
```
Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_main-nav.scss]: We can use the indent variable here.

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_messages.scss [hunk 3]
```diff
@@ -100,3 +100,24 @@
         padding-right: 3em;
     }
 }
+
+// make way for the navbar toggle icon when in mobile state
+@media screen and (max-width: $breakpoint-mobile) {
+    .success {
```
Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_messages.scss]: Can we group these 3 together? Like ``
.success:before,
.error:before,
.warning:before
``

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_messages.scss [hunk 4]
```diff
@@ -100,3 +100,24 @@
...
+        &:before {
+            padding-left: 50px;
```
Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_messages.scss]: We can use the indent variable here.

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss (hunk 1 above)
Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss]: Or maybe even `$mobile-nav-indent` for the case of the avatar.

**Comment by kapito** [comment]
Thanks for taking the time to review it again @jjanssen! I'll work on these changes tomorrow.

File: wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss (hunk 1 above)
Comment (kapito) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss]: Where should the variable be defined? In `_variables.scss` or another place?

Comment (jjanssen) [inline wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_header.scss]: Yup, `_variables.scss` is good.

**Comment by thibaudcolas** [comment]
Hey @jjanssen, let me know if I can help re-reviewing / merging this if you're short on time.

**Comment by jjanssen** [comment]
@thibaudcolas that would be really helpful ❤️  This issue is just awaiting some changes.

**Comment by thibaudcolas** [comment]
@kapito let me know when you think this is ready for testing/review and I'll try to squeeze it before the 1.12 release.

**Comment by thibaudcolas** [comment]
Merged in e008838, thanks @kapito!

I managed to kill this PR and did not get to write my review comments here, so here are a few:

- Moved the `.avatar` overrides with other `.avatar` styles in `wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_formatters.scss`.
- Moved the icon back from the messages area to the header – it's not meant to be within the messages.

**Pull request closed without merging**